src/python/pants/backend/java/target_types.py

```diff
@@ -18,40 +18,68 @@
 from pants.engine.unions import UnionMembership, UnionRule
 
 
-class JavaSources(Sources):
+class JavaSourceField(Sources):
     expected_file_extensions = (".java",)
+    expected_num_files = 1
+
+
+class JavaGeneratorSources(Sources):
+    pass
+    # TODO: do I need anything else here?
```

Comment:
Add `expected_file_extensions = (".java",)`. It duplicates `JavaSourceField`, but that's acceptable duplication - we want `JavaSourceField` to be completely unrelated